docs/deeplearning4j/templates/build-from-source.md

```diff
@@ -12,60 +12,77 @@ weight: 10
+* [ScalNet](https://github.com/deeplearning4j/deeplearning4j/tree/master/scalnet)
+* [PyDL4J](https://github.com/deeplearning4j/deeplearning4j/tree/master/pydl4j)
+* [Jumpy](https://github.com/deeplearning4j/deeplearning4j/tree/master/jumpy)
+* [PyDatavec](https://github.com/deeplearning4j/deeplearning4j/tree/master/pydatavec)
+* [DeepLearning4J](https://github.com/deeplearning4j/deeplearning4j/tree/master/deeplearning4j)
 
-Note that Deeplearning4j is designed to work on most platforms (Windows, OS X, and Linux) and is also includes multiple "flavors" depending on the computing architecture you choose to utilize. This includes CPU (OpenBLAS, MKL, ATLAS) and GPU (CUDA). The DL4J stack also supports x86 and PowerPC architectures.
+Note that Deeplearning4j is designed to work on most platforms (Windows, OS X, and Linux) and it also includes multiple "flavors" depending on the computing architecture you choose to utilize, such as CUDA for GPUs.
 
 ## Prerequisites
 
-Your local machine will require some essential software and environment variables set *before* you try to build and install the DL4J stack. Depending on your platform and the version of your operating system, the instructions may vary in getting them to work. This software includes:
-
-- git
-- cmake (3.2 or higher)
-- OpenMP
-- gcc (4.9 or higher)
-- maven (3.3 or higher)
-
-Architecture-specific software includes:
-
-CPU options:
-
-- Intel MKL
-- OpenBLAS
-- ATLAS
-
-GPU options:
-
-- CUDA
+Your local machine will require some essential softwares and environment variables set *before* you try to build and install DL4J. Depending on your platform and the version of your operating system, the instructions may vary in getting them to work. This required softwares include:
 
+* Git
+* Cmake
+* OpenMP 4.5 supported compiler (Recommended)
+* GCC >= 4.9
+* G++ >= 4.9
+* Java >= 1.7
+* Maven >= 3.3
+* Python Developer Tools
+* CUDA SDK >= 8.0 (Optional - For GPU build)
 
 IDE-specific requirements:
 
-- IntelliJ Lombok plugin
+* Lombok Plugin for IntelliJ or Eclipse
 
 DL4J testing dependencies:
 
-- dl4j-test-resources
+* [dl4j-test-resources](https://github.com/deeplearning4j/dl4j-test-resources) (Needed only for running tests.)
 
 ### Installing Prerequisite Tools
 
-#### Linux
+#### Ubuntu
 
-**Ubuntu**
 Assuming you are using Ubuntu as your flavor of Linux and you are running as a non-root user, follow these steps to install prerequisite software:
 
-```
-sudo apt-get purge maven maven2 maven3
-sudo add-apt-repository ppa:natecarlson/maven3
-sudo apt-get update
-sudo apt-get install maven build-essentials cmake libgomp1
+```bash
+# Installing JDK, Cmake, git, GCC and G++
+sudo apt-get clean -y all
+sudo apt-get update -y
+sudo apt-get upgrade -y
+sudo apt-get install -y cmake git openjdk-8-jdk build-essential
+echo 'export JAVA_HOME=/usr/lib/jvm/java-1.8.0-openjdk-amd64' >> ~/.profile
+source ~/.profile
+
+# Installing Maven
```

Comment:
@printomi yeah I wanted to avoid depending on the versions that Ubuntu was offering. Instead I just wrote a script that's really handy for me. This way the users would also know which version they're going to install beforehand.